### PR TITLE
okhttp: fix incorrect connection-level flow control handling at beginning of connection

### PR DESCRIPTION
**Pull request by chrisschek**
Addressing #6685 

Specifically, this addresses bugs that occur when the `OkHttpChannelBuilder.flowControlWindow(int)` setting is increased from its default value.

Two changes:
1. On starting a connection, ensure the value of `OkHttpChannelBuilder.flowControlWindow(int)` is sent via Settings.INITIAL_WINDOW_SIZE. Also send a WINDOW_UPDATE after Settings to update the connection-level window.
2. Always initialize the `OutboundFlowController` with an initialWindowSize of 65335 bytes per the [http2 spec](https://http2.github.io/http2-spec/#InitialWindowSize) instead of using the inbound window size.

### CODE REVIEW

**Comment by ejona86** [comment]
Thanks for sending this out. I didn't send it out earlier because it _clearly_ deserved some tests. I was on vacation for a bit over a week, thus the delay.

Thank you testing this and confirming the OutboundFlowController argument was garbage.

You'd need to sign the CLA for us to accept the PR. We can work on some tests and make a PR, if that is easiest.

**Comment by chrisschek** [comment]
I signed it

**Comment by chrisschek** [comment]
Hm, I signed the CLA (both on EasyCLA and the [CNCF one linked in the contributing docs](https://identity.linuxfoundation.org/projects/cncf) last week but it doesn't seem to be picking that up.

I took a stab at testing the changes:
1. Was pretty straightforward for the OutboundFlowController argument; the `outboundFlowControl_*` tests seemed redundant after my fix so consolidated them into a single test.
2. For sending INITIAL_WINDOW_SIZE in the first Settings frame, testing was trickier since that code doesn't normally run in the test. Not super happy with my end product having to change source code for the sake of testing, but seemed like a necessary evil.

**Comment by chrisschek** [comment]
Mystery solved – my commits had the wrong email address on them. :)

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java [hunk 1]
```diff
@@ -838,37 +868,34 @@ public void windowUpdateWithInboundFlowControl() throws Exception {
 
   @Test
   public void outboundFlowControl() throws Exception {
-    outboundFlowControl(INITIAL_WINDOW_SIZE);
```
[the remark this answers is not in the document]
Comment (chrisschek) [reply okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java]: Done. I initially thought those cases were already covered by `outboundFlowControlWithInitialWindowSizeChange()`, but realized it wasn't testing where the initial window size is changed before the stream is started.

**Comment by ejona86** [comment]
@chrisschek, thank you! I was on vacation last week so I'm happy that you worked on this.

Sorry you were bit by this silly bug!

**Comment by chrisschek** [comment]
@ejona86 not a problem, happy to contribute :)

**Pull request merged**